pilot/pkg/xds/ads.go

```diff
@@ -215,18 +215,6 @@ func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *C
 
 // StreamAggregatedResources implements the ADS interface.
 func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer) error {
-	// Check if server is ready to accept clients and process new requests.
-	// Currently ready means caches have been synced and hence can build
-	// clusters correctly. Without this check, InitContext() call below would
-	// initialize with empty config, leading to reconnected Envoys loosing
-	// configuration. This is an additional safety check inaddition to adding
-	// cachesSynced logic to readiness probe to handle cases where kube-proxy
-	// ip tables update latencies.
-	// See https://github.com/istio/istio/issues/25495.
-	if !s.IsServerReady() {
```

Comment:
I think it is better to keep this check for safety because you are any way calling CachesSynced method after caches have been synced - It uses the same atomic bool to validate.  If this is removed, there is no value in calling CachesSynced (which may be fine because you are waiting for cache sync before starting any gRPC server). But I would still leave it for safety.

Reply:
The server is not run before the cache synced. And the ready is once for all. So why do we even need this?

Reply:
I agree it theoretically is not needed but am also don't mind keeping it as a redundant check

Reply:
Yeah. I too think it is safe to leave this call as is - it is also check in readiness probe and there is no harm/penalty checking one more time here  (just in case)

Reply:
After a thorough thought, i do think it is necessary in case ads server is called directly without through k8s svc.